.github/workflows/build.yml

```diff
@@ -1,22 +1,34 @@
-# This workflow will install Python dependencies and run tests with several versions of Python
+# This workflow will install Python dependencies and run tests with several versions of Python on MacOS, Ubuntu and Windows
 # When successful, packages are published to test.pypi.org and pypi.org (releases only)
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
 name: Build
 
+env:
+  PYTHON_VERSION: 3.9
+  POETRY_VERSION: 1.1.10
+
 on:
   push:
     branches: [master]
   pull_request:
     branches: [master]
 
 jobs:
-  build:
-    runs-on: ubuntu-latest
+  test-and-package-wheel-archive:
+    name: Build wheel (${{ matrix.python-version }} on ${{ matrix.os }})
+    runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
       matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
         python-version: [3.5, 3.6, 3.7, 3.8, 3.9, 3.10-dev]
+        exclude:
+          # Python 3.10-dev on Windows is excluded for now as this combination:
+          # - has no pywin32 wheel available yet, see https://github.com/mhammond/pywin32/issues/1588 and https://pypi.org/project/pywin32/301/#files
+          # - has no lxml wheel available yet, and building lxml 4.6.3 fails, even with Cython 0.29.24 installed
```

Comment:
Once the first issue is resolved on next pywin32 release, maybe switch from [lxml](https://pypi.org/project/lxml/) to [beautifulsoup4](https://pypi.org/project/beautifulsoup4/) which has no native dependencies.

Reply:
:man_facepalming: My bad, beautifulsoup4 depends on lxml so that won't help here: https://bazaar.launchpad.net/~leonardr/beautifulsoup/bs4/view/386/setup.py#L16